### ISSUE TITLE
Check for sample operations in shader doesn't work with separate samplers

### ISSUE THREAD

**Issue by Rua** [KhronosGroup/Vulkan-ValidationLayers#3623]
The code in `SHADER_MODULE_STATE::IsSpecificDescriptorType` is supposed to check how specific descriptor variables are used. Two of the checks that are performed are for `is_sampler_implicitLod_dref_proj` and `is_sampler_bias_offset`, which are `true` if certain SPIR-V operations are used on a sampled image.

This check works fine if the sampled image is loaded directly from a descriptor with an `OpLoad` or `OpAccessChain`. But it forgets to account for the possibility that the image and sampler were provided as separate bindings. In this case, the ID targeted by the instruction is an `OpSampledImage` instruction, which then targets a separate image and sampler that are each loaded by `OpLoad` and `OpAccessChain`.

**Comment by sfricke-samsung** [comment]
I was going to write up a test to confirm this and now looking into this the 3 VUs for unnormalized coordinates (`VUID-vkCmdDraw-None-02702`, `VUID-vkCmdDraw-None-02703`, `VUID-vkCmdDraw-None-02704`) There seems to be some other issues as well

tl;dr This is a valid issue and I have ran into this similar issue [before](https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/2442) and can look into this
